src/main/java/ea/maps/MapFactory.java

```diff
@@ -2,21 +2,35 @@
 
 import com.fasterxml.jackson.core.type.TypeReference;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.util.ResourceUtils;
 
-import java.io.File;
 import java.io.IOException;
 import java.util.List;
 
 class MapFactory {
 
-    static List<City> getCityListFromJson(String jsonFilePath)
-        throws IOException {
+    static List<City> getCityListFromJson(String countryName) {
 
         ObjectMapper mapper = new ObjectMapper();
 
-        return mapper.readValue(
-            new File(jsonFilePath),
-            new TypeReference<List<City>>() { });
-    }
+        String json;
+        List<City> cityList;
+
+        try {
+
+            json = new String(
+                ResourceUtils.getURL("classpath:" + countryName + ".json")
+                    .openStream()
+                    .readAllBytes()
+            );
 
+            cityList = mapper.readValue(json, new TypeReference<List<City>>() { });
+
+        } catch (IOException ioe) {
+            throw new IllegalStateException(
```

Comment:
Why IllegalStateException? Why not just RuntimeException?

IllegalStateException: Signals that a method has been invoked at an illegal or inappropriate time.

Reply:
I'll catch this in one of the commits for my next pull request.